src/util/script_detection.ts

```
@@ -23,43 +23,71 @@ export function allowsLetterSpacing(chars: string) {
     return true;
 }
 
-export function charAllowsLetterSpacing(char: number) {
-    if (isChar['Arabic'](char)) return false;
-    if (isChar['Arabic Supplement'](char)) return false;
-    if (isChar['Arabic Extended-A'](char)) return false;
-    if (isChar['Arabic Extended-B'](char)) return false;
-    if (isChar['Arabic Presentation Forms-A'](char)) return false;
-    if (isChar['Arabic Presentation Forms-B'](char)) return false;
+/**
+ * Returns a regular expression matching the given script codes, excluding any
+ * code that the execution environment lacks support for in regular expressions.
+ */
+function sanitizedRegExpFromScriptCodes(scriptCodes: Array<string>): RegExp {
+    const supportedPropertyEscapes = scriptCodes.map(code => {
+        try {
+            return new RegExp(`\\p{sc=${code}}`, 'u').source;
+        } catch (e) {
+            return null;
```

When does this fail that this requires try-catch?

---

Unicode keeps adding more scripts every year, and different browsers seem to be updating their Unicode databases at different paces. For example, Firefox 128 recognizes Chorasmian (`Chrs`), but Node v20.11 does not. An unrecognized property escape results in an exception:

> Uncaught SyntaxError: invalid property name in regular expression

We have to swallow this exception without affecting the rest of the regular expression. This check runs only once before we ever test a character against the regular expression.